packages/block-library/src/button/edit.native.js

```diff
@@ -86,118 +87,115 @@ function WidthPanel( { selectedWidth, setAttributes } ) {
-			url: {
-				label: __( 'Button Link URL' ),
-				placeholder: __( 'Add URL' ),
-				autoFocus: true,
-				autoFill: true,
-			},
-			openInNewTab: {
-				label: __( 'Open in new tab' ),
-			},
-			linkRel: {
-				label: __( 'Link Rel' ),
-				placeholder: _x(
-					'None',
-					'Link rel attribute value placeholder'
-				),
-			},
-		};
+	const { editorSidebarOpened, numOfButtons } = useSelect(
+		( select ) => {
+			const { isEditorSidebarOpened } = select( editPostStore );
+			const { getBlockCount, getBlockRootClientId } = select(
+				blockEditorStore
+			);
+			const parentId = getBlockRootClientId( clientId );
+			const blockCount = getBlockCount( parentId );
+			const currentIsEditorSidebarOpened = isEditorSidebarOpened();
+
+			return {
+				editorSidebarOpened: isSelected && currentIsEditorSidebarOpened,
+				numOfButtons: blockCount,
+			};
+		},
+		[ clientId, isSelected ]
+	);
 
-		this.noFocusLinkSettingOptions = {
-			...this.linkSettingsOptions,
-			url: {
-				...this.linkSettingsOptions.url,
-				autoFocus: false,
-			},
-		};
-	}
+	const { closeGeneralSidebar } = useDispatch( editPostStore );
+	const [ maxWidth, setMaxWidth ] = useState( INITIAL_MAX_WIDTH );
+	const [ isLinkSheetVisible, setIsLinkSheetVisible ] = useState( false );
+	const [ isButtonFocused, setIsButtonFocused ] = useState( true );
+	const [ placeholderTextWidth, setPlaceholderTextWidth ] = useState( 0 );
+	const [ borderRadiusUnit, setBorderRadiusUnit ] = useState( valueUnit );
 
-	componentDidMount() {
-		this.onSetMaxWidth();
-	}
+	const richTextRef = useRef();
+	const colors = useSetting( 'color.palette' ) || [];
+	const gradients = useSetting( 'color.gradients' ) || [];
 
-	componentDidUpdate( prevProps, prevState ) {
-		const { isSelected, editorSidebarOpened, parentWidth } = this.props;
-		const { isLinkSheetVisible, isButtonFocused } = this.state;
+	useEffect( () => {
+		onSetMaxWidth();
+	}, [] );
 
-		if ( isSelected && ! prevProps.isSelected ) {
-			this.onToggleButtonFocus( true );
+	useEffect( () => {
+		if ( isSelected ) {
+			onToggleButtonFocus( true );
 		}
+	}, [ isSelected ] );
 
-		if ( prevProps.parentWidth !== parentWidth ) {
-			this.onSetMaxWidth( null, true );
-		}
+	useEffect( () => {
+		onSetMaxWidth( null, true );
+	}, [ parentWidth ] );
 
+	useEffect( () => {
 		// Blur `RichText` on Android when link settings sheet or button settings sheet is opened,
 		// to avoid flashing caret after closing one of them
-		if (
-			( ! prevProps.editorSidebarOpened && editorSidebarOpened ) ||
-			( ! prevState.isLinkSheetVisible && isLinkSheetVisible )
-		) {
-			if ( Platform.OS === 'android' && this.richTextRef ) {
-				this.richTextRef.blur();
-				this.onToggleButtonFocus( false );
+		if ( editorSidebarOpened || isLinkSheetVisible ) {
+			if ( Platform.OS === 'android' && richTextRef?.current ) {
+				richTextRef.current.blur();
+				onToggleButtonFocus( false );
 			}
 		}
+	}, [ editorSidebarOpened, isLinkSheetVisible ] );
```

Comment:
Should we also pass in the `richTextRef?.current` argument here? Since that is also being used by this useEffect?